test/files/pos/t11136_final_in_trait.scala

```diff
@@ -0,0 +1,27 @@
+trait IterableOnce[T]
+
+trait IterableOps[+A, +CC[_]] {
+  def concat[B >: A](suffix: IterableOnce[B]): CC[B] = ???
+}
+
+trait SeqOps[+A, +CC[_]] extends IterableOps[A, CC] {
+  final override def concat[B >: A](suffix: IterableOnce[B]): CC[B] = ???
+}
+
+trait LinearSeqOps[+A, +CC[_]] extends SeqOps[A, CC]
+
+
+trait StrictOptimizedIterableOps[+A, +CC[_]] extends IterableOps[A, CC] {
+  override def concat[B >: A](suffix: IterableOnce[B]): CC[B] = ???
+}
+
+// error reported, but is this really needed?
```

Comment:
when skipping `printMixinOverrideErrors`, we seem to emit perfectly valid bytecode for this example /cc @retronym

Reply:
> Methods of interfaces may have any of the flags in Table 4.6-A set except
ACC_PROTECTED, ACC_FINAL, ACC_SYNCHRONIZED, and ACC_NATIVE (JLS §9.4).

We don't/can't emit default methods as ACC_FINAL. That doesn't mean the error is invalid -- I think it is correctly enforcing Scala language rules.

Reply:
Hmm, now I see your point: given that the override is not user-written, why don't we take advantage of the fact that default methods are known to be non-`ACC_FINAL`.

Reply:
> why don't we take advantage of the fact that default methods are known to be non-`ACC_FINAL`.

That seems kind of crazy to me, I could imagine myself debugging a piece of code for hours, assuming that a given method must be called since it's `final`, only to discover the compiler ignored the language semantics and sneakily added an override while I wasn't looking...

Reply:
I agree it's counter-intuitive to allow the trait definition, and I'm not necessarily saying we must, but as my meta-point: is it that any weirder that you can "fix" the problem later by mixing in the traits in the right order? This, we do all the time to get the desired linearisation order.  I mean, that's the whole reason I'm looking at this bug, because the collections pulled tricks like this. 

I guess you'd say they should not change the linearisation as drastically in subclasses.

Reply:
Also, to be clear. There would be an error in the class of course, because there we know for sure that you're trying to override a final method. We don't know that in the trait, anymore than where a super is going to go (which we acknowledge by having abstract super accessors, but we don't admit that we don't know whether you will actually be overriding a final method or not until the linearisation is decided in a class)